lib/statsd.js

```diff
@@ -354,7 +321,16 @@ Client.prototype.sendMessage = function (message, callback) {
     return;
   }
 
-  if (!this.socket) {
+  const socketWasMissing = !this.socket;
```

Comment:
It **may** be considered a breaking change that I've added an attempt to recover here

Reply:
Good point, thanks.  Not sure yet if this will be a minor or major release, but I think this part by itself would fit in ok with what has been minor releases.